src/Workspaces.cc

```diff
@@ -399,7 +399,23 @@ Workspaces::insert(PWinObj *wo, bool raise)
         }
     }
 
-    _wobjs.insert(it, wo);
+    it = _wobjs.insert(it, wo);
+
+    // Iconify fullscreen windows that may hide the newly raised
```

Comment:
This code really doesn't belong in the insert function, initial suggestion would be to:

```
if (raise) {
    Workspaces::iconifyFullscreenWindows();
}
Workspaces::insert(wo, raise);
```

I guess using a reverse iterator in the iconify could get some of the extra iteration out of the way.

Reply:
Hah! Nice idea about reverse iterator. I actually had it as a separate function first, but I had to make a copy of the iterator and did like it very much.